Extensions/BBText/JsExtension.js

```diff
@@ -0,0 +1,601 @@
+        wordWrap: true,
+      })
+    );
+
+    objectBBText.updateInitialInstanceProperty = function(
+      objectContent,
+      instance,
+      propertyName,
+      newValue,
+      project,
+      layout
+    ) {
+      return false;
+    };
+    objectBBText.getInitialInstanceProperties = function(
+      content,
+      instance,
+      project,
+      layout
+    ) {
+      var instanceProperties = new gd.MapStringPropertyDescriptor();
+      return instanceProperties;
+    };
+
+    const object = extension
+      .addObject(
+        'BBText',
+        _('BBText'),
+        _(
+          'Displays a rich text label using BBCode markup (allowing to set parts of the text as bold, italic, use different colors and shadows).'
+        ),
+        'JsPlatform/Extensions/bbcode32.png',
+        objectBBText
+      )
+      .setIncludeFile('Extensions/BBText/bbtextruntimeobject.js')
+      .addIncludeFile('Extensions/BBText/bbtextruntimeobject-pixi-renderer.js')
+      .addIncludeFile('Extensions/BBText/dist/pixi-multistyle-text.umd.js');
+
+    // Utility function to add both a setter and a getter to a property from a list. Useful for setting multiple generic properties
+    const addSettersAndGettersToObjectHelper = (
+      gdObject,
+      properties,
+      objectName
+    ) => {
+      properties.forEach(property => {
+        property.type = property.type === 'boolean' ? 'yesorno' : property.type;
+
+        // Add all the generic GETTERS
+        if (property.type === 'number') {
+          gdObject
+            .addExpression(
+              `Get${property.functionName}`,
+              _(`Get ${property.description}`),
+              _(`Get ${property.description}`),
+              _(''),
+              '',
+              `${property.iconPath}.png`,
+              `${property.iconPath}.png`
+            )
+            .addParameter(
+              'object',
+              _(`${objectName} object`),
+              objectName,
+              false
+            )
+            .getCodeExtraInformation()
+            .setFunctionName(`get${property.functionName}`);
+        } else {
+          gdObject
+            .addStrExpression(
+              `Get${property.functionName}`,
+              _(`Get ${property.description}`),
+              _(`Get ${property.description}`),
+              _(''),
+              '',
+              `${property.iconPath}.png`,
+              `${property.iconPath}.png`
+            )
+            .addParameter(
+              'object',
+              _(`${objectName} object`),
+              objectName,
+              false
+            )
+            .getCodeExtraInformation()
+            .setFunctionName(`get${property.functionName}`);
+        }
+
+        // Add all the generic SETTERS
+        if (property.type === 'number' || property.type === 'string') {
+          gdObject
+            .addAction(
+              `Set${property.functionName}`,
+              _(`Set ${property.description}`),
```

Comment:
I'm afraid `_(...)` won't work with string interpolation :/ The translation tools won't work with this (they might in GD React components), but not here because the translation is only done on strings.

Reply:
That's very interesting :) it would be incredible if the util approach even supports translations